### ISSUE TITLE
LPub3D 2.3.13 and 2.4 not showing times indicator in certain instances

### ISSUE THREAD

**Issue by Landcross** [trevorsandy/lpub3d#480]
As the title says; LPub3D 2.3.13 is not showing the times indicator in specific instances. I have no idea what specific instances, I just know that the model I'm working on doesn't show the times indicator for a few submodels.

For example:
![image](https://user-images.githubusercontent.com/12005521/98351314-d33b8b00-201c-11eb-8d43-4048198f003c.png)

This the last page of this submodel and on the next page, it's inserted twice. However, I don't see any 2X on the page.
And a similar thing is happening with other submodels in the same model.

However, other submodels do show the times indicator. For example:
![image](https://user-images.githubusercontent.com/12005521/98351520-1b5aad80-201d-11eb-8426-591da238033a.png)

I tested it in both 2.3.13 and 2.4 and the behavior is the same in both versions.

I can't share the model here in public, but I'd be able to send it to you in private if that's possible.

**Comment by trevorsandy** [comment]
Thank you for reporting this behaviour. I will take a look.

You can send your example model file to my email address available in the 'About' dialogue and let me know when you send it so I can check.

Cheers,

**Comment by Landcross** [comment]
Hold up.... I think I found out something. I was using 'continuous step numbers' for the first time and when checking out the final pdf I noticed that the step numbers were all over the place; they weren't right in certain places. So, I switched it off and generated a new pdf. I was checking out the step numbers and noticed this:

![image](https://user-images.githubusercontent.com/12005521/98436889-2544e500-20df-11eb-827c-5fc0913a60db.png)

The 2x is now visible, while it wasn't before (see image in the original post).

I just reopened LPub3D and I can confirm: switching between continuous step numbers makes certain 2x's appear and disappear.

So, there seem to be two issues:

- Continuous step mode makes certain 2x's disappear (in this model at least) (tested both 2.3.13 and 2.4)
- Continuous step mode makes a mess of certain step numbers (in this model at least) (so far only tested in 2.3.13, though I'm 95% it's also in 2.4, I'll have to test it out later)

If I can find some time tonight, I'll do some experiments and see if I can reproduce the above stuff with simple models.

**Comment by trevorsandy** [comment]
Thank you for the update. I’ll wait for your further testing.

On the rounded corners issue (#473), I noticed the corners of the PLI background appear to be more uniform in your latest screenshots than the initial ones and those in #473. Did something change ?

Cheers,

**Comment by Landcross** [comment]
> On the rounded corners issue (#473), I noticed the corners of the PLI background appear to be more uniform in your latest screenshots than the initial ones and those in #473. Did something change ?

Haha, well spotted! I'm afraid nothing changed except that because of that rounding issue (and the instability) I had to revert back to 2.3.13 (which doesn't have the rounding issue) to finish the model I am working on.

**Comment by trevorsandy** [comment]
So the rounding behaviour was likely introduced in 2.4.0. This is good information. I'll take a look.

Cheers,

**Comment by Landcross** [comment]
Ok, both the above issues I cannot reproduce with a simple model, but I can reproduce them with the specific model I'm working on, in both version 2.3.13 and 2.4. I will email you the model and the details.

**Comment by trevorsandy** [comment]
Very well. Thank you.

Cheers,

**Comment by trevorsandy** [comment]
Corrected. 

The instance count is set to display at the last step of the submodel. With continuous step number enabled, the last step of a submodel was not being correctly identified on multi-step pages so the instance count was not displayed. Basically, the last step was set as the step passed to the 'draw page' call which is ok for single-step pages but is the first step in a multi-step group. This behaviour has been corrected.

![Screenshot - 14_11_2020 , 09_41_28](https://user-images.githubusercontent.com/13388970/99143522-65b3de00-265e-11eb-9a99-01f6e158d330.png)

Other related bahaviour from your email:

_Irregular step numbers when continuous step numbers is enabled:_
I did not see any obvious irregularities so I'll wait for you to create an issue and example model.

_The colors in 2.4 are lighter than in 2.3.13:_
This may be the result of changes in the LDConfig.ldr file. LPub3D downloads the latest LDraw archive libraries at application build so it is entirely possible to have different hues on some colors between releases.

_Part 98138 is missing top-edge:_
Part 98138 is beveled at the cylinder and top disc intersection. There is no edge to display as a line. 

Here are some other renderings which correctly show, as does the LPub3D shot, 98138 as it was designed.

![Screenshot - 14_11_2020 , 09_49_54](https://user-images.githubusercontent.com/13388970/99143634-5f723180-265f-11eb-9afc-6d605f357fb6.png)

![Screenshot - 14_11_2020 , 09_51_11](https://user-images.githubusercontent.com/13388970/99143644-6d27b700-265f-11eb-8cc9-e34e39992603.png)

Cheers,
